### ISSUE TITLE
[RFE] Tool to display current configuration

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1832]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1828
- Created at 2013-03-07 15:11:06 by [rcritten](https://pagure.io/user/rcritten)
- Closed at 2020-03-24 14:12:09 as wontfix
- Assigned to nobody
---

It would be nice to have a tool like Samba's testparm command to display the current configuration. The ad and ipa providers hide a lot of complexity by pre-defining a lot of default attributes, but you don't necessarily know exactly what sssd is going to do because a lot of configuration isn't shown in the config file.

sgallagh suggested that this should be easily accessible via the python bindings.

### Comments

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2013-03-07 15:58:37](https://pagure.io/SSSD/sssd/issue/1828#241267)**

I looked into this today, and it's really not as easy as it sounds. The SSSDConfig API isn't built to show the internal defaults, it's really only designed for handling manual configuration and overrides.

There are a couple problems here:
1) The SSSDConfig API data files do not have the defaults for all entries. We would have to mirror the contents of the various ldap_opt type files (as well as all of the places that we only list the default in the source and manpage). Fixing this would pretty much require completing ticket #1386.

2) Not all options have pure defaults. We have multiple examples of options whose default is to use the value of another option (which may or may not have been set). For example, {{{ldap_user_search_base}}} defaults to the value of {{{ldap_search_base}}}, which may or may not have been set. Furthermore, {{{ldap_search_base}}} defaults to the value detected from the RootDSE at connection time, which we cannot know ahead of time in a tool. Other examples include the {{{ipa_domain}}} option which automatically sets the default for {{{krb5_realm}}} and {{{dns_discovery_domain}}}.

In short, I'm inclined to say that this ticket is unsolvable as a tool, though we can (and should) do a better job of reporting every configuration value in the debug logs at {{{SSSDBG_CONF_SETTINGS}}} when it is initially determined by the code. This way an admin can at least view the debug logs to get this information.

blockedby:  => 1386
cc:  => chudson@redhat.com, sgallagh
component: SSSD => sss_tools

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2013-03-07 16:06:30](https://pagure.io/SSSD/sssd/issue/1828#241268)**

What about an option to sssd itself to say "read the config, get things set up, but instead of forking off the daemons just print the config and quit"?

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2013-03-07 16:14:13](https://pagure.io/SSSD/sssd/issue/1828#241269)**

Still won't work for the reasons above. Some of the defaults differ based on the value of other options, which provider it is, etc. Furthermore, we have options that may not be configured until we read the RootDSE.

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2013-03-07 16:36:31](https://pagure.io/SSSD/sssd/issue/1828#241270)**

We can add internal sbus calls to query the various backends, but I do
not know if the work needed is justified for now.

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2013-03-07 16:40:20](https://pagure.io/SSSD/sssd/issue/1828#241271)**

How about a log level that logs the config on startup. 

The use case for this, unlike Samba, isn't "let me test the configuration" it is more "what the heck is the configuration being used since there are a lot of defaults". So I think having any way to see what all the options are is acceptable.

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2013-03-07 16:51:19](https://pagure.io/SSSD/sssd/issue/1828#241272)**

We already log the options on startup but as Stephen said above, we should make sure these logs use a predictable log level. Currently it's a mix of what seemed like a good idea at the time.

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2013-03-08 01:41:12](https://pagure.io/SSSD/sssd/issue/1828#241273)**

IMO the right solution would be to keep an object in memory that would reflect currently resolved configuration. I should be built dynamically as things are read from config DB resolved. Then this object should be queried by the tool probably over SBUS.

Some work. Probably a good candidate for a thesis.

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2013-03-10 23:05:28](https://pagure.io/SSSD/sssd/issue/1828#241274)**

Fields changed

cc: chudson@redhat.com, sgallagh => chudson@redhat.com, sgallagh, chudson

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2013-03-14 14:55:17](https://pagure.io/SSSD/sssd/issue/1828#241275)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.12 beta
rhbz:  => todo

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2013-12-02 20:36:52](https://pagure.io/SSSD/sssd/issue/1828#241276)**

Linked to Bugzilla bug: https://bugzilla.redhat.com/show_bug.cgi?id=879333 (*RHEL RFE*)

rhbz: todo => [https://bugzilla.redhat.com/show_bug.cgi?id=879333 879333]

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2013-12-02 20:38:25](https://pagure.io/SSSD/sssd/issue/1828#241277)**

Under this ticket we should provide a way to report all domain information statically and dynamically configured for SSSD.

changelog:  => 
review:  => 0

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2014-03-07 15:44:37](https://pagure.io/SSSD/sssd/issue/1828#241278)**

Fields changed

keywords:  => Status

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2014-03-07 15:44:55](https://pagure.io/SSSD/sssd/issue/1828#241279)**

Fields changed

blockedby: 1386 => #1386

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2015-12-11 19:33:18](https://pagure.io/SSSD/sssd/issue/1828#241280)**

Linked to Bugzilla bug: https://bugzilla.redhat.com/show_bug.cgi?id=1290500 (*Red Hat Enterprise Linux 7*)

rhbz: [https://bugzilla.redhat.com/show_bug.cgi?id=879333 879333] => [https://bugzilla.redhat.com/show_bug.cgi?id=879333 879333], [https://bugzilla.redhat.com/show_bug.cgi?id=1290500 1290500]

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2016-02-16 13:34:34](https://pagure.io/SSSD/sssd/issue/1828#241281)**

Pavel plans on working on a status utility, at least some of the information could be exposed over D-Bus.

mark:  => 0
owner: somebody => preichl
sensitive:  => 0

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2016-04-18 18:56:21](https://pagure.io/SSSD/sssd/issue/1828#241282)**

Fields changed

design:  => https://fedorahosted.org/sssd/wiki/DesignDocs/sssctl

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2016-06-10 14:29:44](https://pagure.io/SSSD/sssd/issue/1828#241283)**

Fields changed

owner: preichl => pbrezina

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2016-06-27 10:54:51](https://pagure.io/SSSD/sssd/issue/1828#241284)**

We have the sssctl tool, but this functionality is not implemented yet.

milestone: SSSD 1.14 beta => NEEDS_TRIAGE

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2016-07-08 15:40:14](https://pagure.io/SSSD/sssd/issue/1828#241285)**

Fields changed

rhbz: [https://bugzilla.redhat.com/show_bug.cgi?id=879333 879333], [https://bugzilla.redhat.com/show_bug.cgi?id=1290500 1290500] => [https://bugzilla.redhat.com/show_bug.cgi?id=879333 879333]

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2016-07-13 13:44:11](https://pagure.io/SSSD/sssd/issue/1828#241286)**

The sssctl tool already can do a lot of things, like displaying the servers that were discovered or displaying the cache entry timestamps. This functionality, however, is not implemented yet and should be implemented in one of the next releases.

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2016-07-13 13:55:28](https://pagure.io/SSSD/sssd/issue/1828#241287)**

Ticket has been cloned to Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1356094

rhbz: [https://bugzilla.redhat.com/show_bug.cgi?id=879333 879333] => [https://bugzilla.redhat.com/show_bug.cgi?id=879333 879333], [https://bugzilla.redhat.com/show_bug.cgi?id=1356094 1356094]

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2016-07-13 13:56:07](https://pagure.io/SSSD/sssd/issue/1828#241288)**

Fields changed

rhbz: [https://bugzilla.redhat.com/show_bug.cgi?id=879333 879333], [https://bugzilla.redhat.com/show_bug.cgi?id=1356094 1356094] => [https://bugzilla.redhat.com/show_bug.cgi?id=1356094 1356094]

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2016-07-14 17:33:35](https://pagure.io/SSSD/sssd/issue/1828#241289)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.16 beta

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2017-02-06 15:42:18](https://pagure.io/SSSD/sssd/issue/1828#241290)**

Fields changed

milestone: SSSD Future releases (no date set yet) => SSSD 1.15.3

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2017-02-24 15:04:00](https://pagure.io/SSSD/sssd/issue/1828#241291)**

**Metadata Update from @rcritten**:
- Issue assigned to pbrezina
- Issue marked as depending on: #1386
- Issue set to the milestone: SSSD 1.15.3

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2017-03-15 11:47:12](https://pagure.io/SSSD/sssd/issue/1828#431462)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch reset
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset
- Issue close_status updated to: None
- Issue set to the milestone: SSSD 1.15.4 (was: SSSD 1.15.3)

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2017-08-18 16:59:20](https://pagure.io/SSSD/sssd/issue/1828#458789)**

**Metadata Update from @jhrozek**:
- Custom field blocking adjusted to 1386
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue tagged with: cleanup-future

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2017-08-23 17:11:54](https://pagure.io/SSSD/sssd/issue/1828#459941)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue **un**tagged with: cleanup-future
- Issue set to the milestone: SSSD Future releases (no date set yet) (was: SSSD 1.15.4)

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2019-04-01 12:16:34](https://pagure.io/SSSD/sssd/issue/1828#563605)**

**Metadata Update from @pbrezina**:
- Assignee reset

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2020-03-13 14:13:46](https://pagure.io/SSSD/sssd/issue/1828#633175)**

**Metadata Update from @thalman**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue tagged with: Canditate to close, bugzilla

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2020-03-24 14:12:08](https://pagure.io/SSSD/sssd/issue/1828#636678)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [rcritten](https://pagure.io/user/rcritten) at [2020-03-24 14:12:09](https://pagure.io/SSSD/sssd/issue/1828#636679)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
